### PR TITLE
Port testSuite/../niocharset to JUnit and cross compile. 

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

File: test-suite/shared/src/test/scala/org/scalajs/testsuite/niocharset/CharsetTest.scala [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    assertEquals("UTF-16LE", Charset.forName("UnicodeLittleUnmarked").name())
+
+    assertEquals("UTF-16", Charset.forName("UTF-16").name())
+    assertEquals("UTF-16", Charset.forName("Utf_16").name())
+    assertEquals("UTF-16", Charset.forName("unicode").name())
+    assertEquals("UTF-16", Charset.forName("UnicodeBig").name())
+
+    if (1.0.toString != "1") {
+      // bug #2040 - not resolved
+      expectThrows(classOf[UnsupportedCharsetException], Charset.forName("UTF_8"))
```
Comment (nicolasstucki) [inline test-suite/shared/src/test/scala/org/scalajs/testsuite/niocharset/CharsetTest.scala]: Test for issue #2040.

File: test-suite/shared/src/test/scala/org/scalajs/testsuite/niocharset/BaseCharsetTest.scala [hunk 2]
```diff
@@ -14,14 +14,13 @@ import scala.util._
...
-import js.JSConverters._
+import org.junit.Assert._
 
-import org.scalajs.jasminetest.JasmineTest
-
-class BaseCharsetTest(val charset: Charset) extends JasmineTest {
+class BaseCharsetTest(val charset: Charset) {
   import BaseCharsetTest._
 
+  protected def inJVM: Boolean = 1.0.toString != "1"
```
Comment (gzm0) [inline test-suite/shared/src/test/scala/org/scalajs/testsuite/niocharset/BaseCharsetTest.scala]: Is this really how we are officially going to test whether we run in a JVM? *sadface*

Comment (sjrd) [inline test-suite/shared/src/test/scala/org/scalajs/testsuite/niocharset/BaseCharsetTest.scala]: Well I guess we could put that test once and for all in `testsuite.Util` or something. But yes, that's our most reliable test. Or we make it a non-shared `object Platform { final val isJVM = false }` in `js/` and `= true` in `jvm/`.

Comment (nicolasstucki) [inline test-suite/shared/src/test/scala/org/scalajs/testsuite/niocharset/BaseCharsetTest.scala]: I will do that. 
This should only be used in cases where there is a bug found during the migration of tests. I would assume that once those bugs are fixed we should remove this utility and never use it again.

Comment (gzm0) [inline test-suite/shared/src/test/scala/org/scalajs/testsuite/niocharset/BaseCharsetTest.scala]: I think it is definitely worth factoring in a utility class, even if it's only for readability.

**Comment by nicolasstucki** [comment]
There is still a test that fails on malformed inputs in java 6 and 7.

**Comment by sjrd** [comment]
See #1538

**Comment by nicolasstucki** [comment]
Will need a rebase on #2043.

File: test-suite/shared/src/test/scala/org/scalajs/testsuite/niocharset/USASCIITest.scala [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    // Simple tests
+
+    testDecode(bb"48 65 6c 6c 6f")(cb"Hello")
+    testDecode(bb"42 6f 6e 6a 6f 75 72")(cb"Bonjour")
+
+    testDecode(bb"00 01 0a 10 20")(cb"\u0000\u0001\u000a\u0010 ")
+    testDecode(bb"7f 7f")(cb"\u007f\u007f")
+
+    // Bit 7 is ignored, giving the same results as above
+    if (1.0.toString == "1") {
```
Comment (sjrd) [inline test-suite/shared/src/test/scala/org/scalajs/testsuite/niocharset/USASCIITest.scala]: `executingInJS`

File: test-suite/shared/src/test/scala/org/scalajs/testsuite/util/Platform.scala [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package org.scalajs.testsuite.util
+
+object Platform {
+  /** Returns `true` if and only if the code is executing on a JVM.
+   *  Note: Returns `false` when executing on any JS VM.
+   */
+  lazy val executingInJVM: Boolean = {
```
Comment (sjrd) [inline test-suite/shared/src/test/scala/org/scalajs/testsuite/util/Platform.scala]: This should not be there.

File: test-suite/shared/src/test/scala/org/scalajs/testsuite/niocharset/CharsetTest.scala [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    assertTrue(Charset.isSupported("US-ASCII"))
+    assertTrue(Charset.isSupported("Default"))
+    assertTrue(Charset.isSupported("utf-8"))
+    assertTrue(Charset.isSupported("UnicodeBigUnmarked"))
+    assertTrue(Charset.isSupported("Utf_16le"))
+    assertTrue(Charset.isSupported("UTF-16"))
+    assertTrue(Charset.isSupported("unicode"))
+
+    assertFalse(Charset.isSupported("this-charset-does-not-exist"))
+
```
Comment (sjrd) [inline test-suite/shared/src/test/scala/org/scalajs/testsuite/niocharset/CharsetTest.scala]: Spurious blank line.

**Comment by sjrd** [comment]
That's all.

**Comment by sjrd** [comment]
LGTM

**Pull request merged**